test/connection_test.rb

```diff
@@ -450,6 +450,16 @@ def test_default_connection_options_as_hash_with_instance_connection_options
     assert_equal 1, conn.options.open_timeout
   end
 
+  def test_default_connection_options_persist_with_an_instance_overriding
+    Faraday.default_connection_options.request.timeout = 10
+    conn = Faraday.new 'http://nigiri.com/bar'
+
+    other = Faraday.new :url => 'https://sushi.com/foo'
```

Comment:
I have found that this test will pass if the `sushi.com` url here is passed a string argument rather than in this hash form, but fails when passed as seen here.